util/webhook/webhook.go

```diff
@@ -143,20 +188,36 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}, header webhooks.
 }
 
 func (a *ArgoCDWebhookHandler) Handler(w http.ResponseWriter, r *http.Request) {
-	event := r.Header.Get("X-GitHub-Event")
-	if len(event) > 0 {
-		a.githubHandler.ServeHTTP(w, r)
-		return
-	}
-	event = r.Header.Get("X-Gitlab-Event")
-	if len(event) > 0 {
-		a.gitlabHandler.ServeHTTP(w, r)
+
+	var payload interface{}
+	var err error
+
+	switch {
+	//Gogs needs to be checked before Github since it carries both Gogs and (incompatible) Github headers
+	case r.Header.Get("X-Gogs-Event") != "":
+		payload, err = a.gogs.Parse(r, gogs.PushEvent)
+		break
+	case r.Header.Get("X-GitHub-Event") != "":
+		payload, err = a.github.Parse(r, github.PushEvent)
+		break
+	case r.Header.Get("X-Gitlab-Event") != "":
+		payload, err = a.gitlab.Parse(r, gitlab.PushEvents, gitlab.TagEvents)
+		break
+	case r.Header.Get("X-Hook-UUID") != "":
+		payload, err = a.bitbucket.Parse(r, bitbucket.RepoPushEvent)
+		break
+	case r.Header.Get("X-Event-Key") != "":
+		payload, err = a.bitbucketserver.Parse(r, bitbucketserver.RepositoryReferenceChangedEvent)
+		break
+	default:
+		log.Debug("Ignoring unknown webhook event")
```

Comment:
probably should we `Warn`?

Reply:
Debug is definitely to low. I used to be confused about this first time I tried to setup Bitbucket Server (not yet implemented by then) to send webhooks to ArgoCD. In the mean time, I think we would like to avoid spam in case of "DOS" on the `/api/webhook` (if public).

I would suggest a log.Info here.